File: Pkdd.Models/Help/IssueUserInfo.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Text;

namespace Pkdd.Models.Help
{
    [Owned]
    public class IssueUserInfo
    {
        public int UserId { get; set; }

        public string Name { get; set; }

        public string Email { get; set; }

        public string MainRole { get; set; }
    }
}

File: Pkdd/pkdd-ng-client/src/app/help/models/issue-user-info.ts
import { PkddUser } from './../../models/auth/pkdd-user';
import { PkddRoles } from 'src/app/models/auth/pkdd-roles.enum';
export class IssueUserInfo {
    userId: number;
    name: string;
    email: string;
    mainRole: string;
    constructor(user: PkddUser) {
        this.userId = user.id;
        this.name = user.name;
        this.email = user.email;
        user.roles.includes('admin') ? this.mainRole = 'admin' :
            user.roles.includes('tech') ? this.mainRole = 'tech' :
                user.roles.includes('expert') ? this.mainRole = 'expert' : this.mainRole = 'undefined';
    }
}
